### PR TITLE
FIX: Remove loading spinner after image export

### LINKED ISSUE #470: Remove spinner after an export is complete
**Please note we are preparing to participate in Google Code-in, and have reserved this issue for GCI participants - but we'd love to have your help with another one! Please check out https://code.publiclab.org to see more.**

# Problem

LDI provides the ability to export multiple images. 
Demo it here: https://publiclab.github.io/Leaflet.DistortableImage/examples/select.html
by 
- `ctrl + click`ing on one or more images. 
- A single toolbar for all of them will appear in the topleft corner. 
- Click the top icon to begin export
- Open your developer console to see updates on the status

While the export is in progress a spinner icon appears, but it does not yield back to the original export icon after completion. 

We would like it to do so. 

# Solution:

**Relevant information:**

- https://github.com/publiclab/Leaflet.DistortableImage/blob/main/src/edit/actions/ExportAction.js - this is the file for the export action. In `addHooks` we
   - `if (edit instanceof L.DistortableImage.Edit) { edit._getExport(); }` Single image instances have a handler class `L.DistortableImage.Edit` that triggers a single image export `edit._getExport();` (you can test this one by clicking on any image and hitting the export icon on its attached toolbar). Multiple images have a separate handler `L.DistortableCollection.Edit` that calls the multiple export function, `edit.startExport()`, which you demo'd earlier. We check which one to call.
  - if its multiple, follows are method that switch the action's icon to the spinner, and add the `loader` class to it to animate it's spin via css. 
  - Then a call to the multiple export function `edit.startExport()`

https://github.com/publiclab/Leaflet.DistortableImage/blob/f5378980b8712bc8b95faa1d9ce26e2fda5d6cd8/src/edit/actions/ExportAction.js#L24-L35

- https://github.com/publiclab/Leaflet.DistortableImage/blob/main/src/edit/DistortableCollection.Edit.js 
  - In `startExport`, follow the logic to find the relevant lines where the export completes, and use this knowledge to remove the loader icon and render the original export icon again.

https://github.com/publiclab/Leaflet.DistortableImage/blob/f5378980b8712bc8b95faa1d9ce26e2fda5d6cd8/src/edit/DistortableCollection.Edit.js#L194-L257
 

- We use SVG icons, with a simple system for rendering the icons in JS dynamically. Follow the pattern of:
```
# hint - we will want 'get_app' back
 L.IconUtil.toggleXlink(this._link, 'get_app', 'spinner'); 

# toggles the tooltip
 L.IconUtil.toggleTitle(this._link, 'Export Images', 'Loading...'); 

# adds a css spinner class
 L.IconUtil.addClassToSvg(this._link, 'loader'); 
```
to reverse it. 

- This is enough to solve this but if you want to understand more about our SVG process: https://github.com/publiclab/Leaflet.DistortableImage/wiki/SVG-Icon-System

##  💬   Get help
If you need any help - here are some options: 
- Comment below
- Join our gitter chat at https://gitter.im/publiclab/publiclab

Labeled `hard` because requires a deeper understanding of LDI's moving parts.

### PR DESCRIPTION
**Pull request by VladimirMikulic**
In the old UI, the spinner would continue spinning after the images have been
exported. Now it's fixed.

Resolves #470 

@sashadev-sky could you review this, please?

### CODE REVIEW

File: src/edit/DistortableCollection.Edit.js [hunk 1]
```diff
@@ -209,6 +210,10 @@ L.DistortableCollection.Edit = L.Handler.extend({
       }
       if (data.status === 'complete') {
         clearInterval(updateInterval);
+
+        L.IconUtil.toggleXlink(link, 'get_app', 'spinner');
+        L.IconUtil.toggleTitle(link, 'Export Images', 'Loading...');
+        L.DomUtil.removeClass(link.firstChild, 'loader');
       }
       if (data.status === 'complete' && data.jpg !== null) {
```
Comment (chen-robert) [inline src/edit/DistortableCollection.Edit.js]: You could probably move the `data.jpg !== null` condition inside the `data.status === 'complete'` to make the code cleaner. 

Something like

```javascript
if (data.status === 'complete') {
        L.IconUtil.toggleXlink(link, 'get_app', 'spinner');
        L.IconUtil.toggleTitle(link, 'Export Images', 'Loading...');
        L.DomUtil.removeClass(link.firstChild, 'loader');

        if (data.jpg !== null) {
    
```

**Comment by sashadev-sky** [comment]
@chen-robert @VladimirMikulic SORRY should have mentioned this. The exporter never actually completes locally, it needs to be up on github pages. For my testing, I use https://sasha.mapknitter.org/examples/select.html. I added @VladimirMikulic work there and it works, you can check it out!

The exporter is really the only thing you would need you own page for, but if you plan on helping out with it you might want to get one setup! You will need a custom domain which I think @jywarren can provide you guys with, and I'll walk you through how to do it. Let me know!

**Comment by VladimirMikulic** [comment]
@sashadev-sky I wrapped `startExport` function inside of a promise and tested it.
It works as expected.

**Comment by VladimirMikulic** [comment]
@sashadev-sky I would be thankful if you could review this.

**Comment by VladimirMikulic** [comment]
@sashadev-sky polyfill added!

**Comment by VladimirMikulic** [comment]
@sashadev-sky I missed it, sorry.

**Comment by SidharthBansal** [comment]
@chen-robert @jywarren can you please review?

**Comment by VladimirMikulic** [comment]
@sashadev-sky ?

**Comment by jywarren** [comment]
This is great, can you share a gif of this working, or a link to it running in a GitHub pages branch? Thanks!

**Comment by VladimirMikulic** [comment]
@jywarren @sashadev-sky here is the showcase video: https://streamable.com/ye1oz

**Comment by jywarren** [comment]
This is wonderful! Thanks so much, great work!

**Comment by VladimirMikulic** [comment]
@jywarren could you also approve the task on GCI, so I can move on?
Thanks.

**Comment by SidharthBansal** [comment]
Pls give me link. I will approve it

**Comment by VladimirMikulic** [comment]
Thank you @SidharthBansal.

**Comment by sashadev-sky** [comment]
Sorry for the wait! I will review shortly

**Pull request merged**